app/services/appointment_notification_service.rb

```diff
@@ -25,7 +25,7 @@ def eligible_appointments(appointments, communication_type)
     end
 
     def eligible_for_sending_sms?(appointment, communication_type)
-      (not appointment.previously_communicated_via?(communication_type)) && appointment.patient&.phone_number?
+      (not appointment.previously_communicated_via?(communication_type)) && appointment.patient&.mobile_number?
```

Comment:
Consider scoping the Appointments to also only be included if they patient has a mobile based phone number, in this service.